src/Moq/ISetup.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2007, Clarius Consulting, Manas Technology Solutions, InSTEDD.
+// All rights reserved. Licensed under the BSD 3-Clause License; see License.txt.
+
+using System.Linq.Expressions;
+
+namespace Moq
+{
+	/// <summary>
+	///   A setup configured on a mock.
+	/// </summary>
+	public interface ISetup
+	{
+		/// <summary>
+		///   The setup expression.
+		/// </summary>
+		LambdaExpression Expression { get; }
+
+		/// <summary>
+		///   Gets whether this setup is conditional.
+		/// </summary>
+		bool IsConditional { get; }
+
+		/// <summary>
+		///   Gets whether this setup was matched by at least one invocation on the mock.
+		/// </summary>
+		bool IsMatched { get; }
+
+		/// <summary>
+		///   Gets whether this setup is overridden (i.e. shadowed by a more recent setup having the same expression).
+		/// </summary>
+		bool IsOverridden { get; }
+
+		/// <summary>
+		///   Gets whether this setup is verifiable.
+		/// </summary>
+		bool IsVerifiable { get; }
+
+		/// <summary>
+		///   Checks whether this setup is a part of a composite setup.
+		///   If so, the composite setup of which this one is a part is returned via the <see langword="out"/> parameter <paramref name="originalSetup"/>.
+		///   <para>
+		///     (Composite setups are those with expressions where several member accesses or calls are chained together.
+		///     They are a much more accurate reflection of the original setup expressions you will find in your own code,
+		///     whereas the setups returned by <see cref="Mock.Setups"/> are more "technical" in nature:
+		///     They only reflect the part of an original setup expression that is of immediate concern to the mock.)
+		///   </para>
+		/// </summary>
+		/// <param name="originalSetup">
+		///   If this setup is part of a composite setup,
+		///   this <see langword="out"/> parameter will be set to the latter.
+		/// </param>
+		/// <returns>
+		///   <see langword="true"/> if this setup is a part of a composite setup;
+		///   otherwise, <see langword="false"/>.
+		/// </returns>
+		/// <seealso cref="ICompositeSetup"/>
+		/// <seealso cref="ReturnsInnerMock(out Mock)"/>
+		bool IsPartOfCompositeSetup(out ICompositeSetup originalSetup);
+
+		/// <summary>
+		///   Checks whether this setup is known to return a mocked object.
+		///   If so, the corresponding <see cref="Mock"/> (the so-called "inner mock") is returned via the <see langword="out"/> parameter <paramref name="innerMock"/> instance.
+		/// </summary>
+		/// <param name="innerMock">
+		///   If this setup is known to return a mocked object,
+		///   this <see langword="out"/> parameter will be set to the corresponding <see cref="Mock"/> instance.
+		/// </param>
+		/// <returns>
+		///   <see langword="true"/> if this setup is known to return a mocked object;
+		///   otherwise, <see langword="false"/>.
+		/// </returns>
+		bool ReturnsInnerMock(out Mock innerMock);
```

Comment:
Same note on two properties -- naming makes sense, but why not `Mock? ReturnedInnerMock` or something similar?

Reply:
A simple property `Mock InnerMock { get; }` (which can also return `null`) would be simplest, I'm undecided whether it's a good idea to make this a property since there's possibly a bit of computation going on.

Reply:
Sure -- `Mock? GetInnerMock()` also sounds fine/clear.